_posts/2023-07-10-trino-fest-2023-redis.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: "Redis & Trino - Real-time indexed SQL queries (new connector)"
+author: "Allen Terleto, Julien Ruaux, Ryan Duan"
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-fest-2023/Redis.png
+show_pagenav: false
+---
+
+Ever since the pandemic, the world has changed. The need for a digital first
+economy rose dramatically during the lockdowns. It has become clear that
+digital's impact on future businesses will be more and more important. As Redis'
+Field CTO Allen Terleto said during their talk from [Trino Fest 2023]({%
+post_url 2023-06-20-trino-fest-2023-recap %}), "In a digital first economy, data
+is the lifeblood of the organization, which makes the databases the heart of
+enterprise architectures". Redis, one of the most popular open source projects,
+is an in-memory data structure store, used as a distributed, in-memory key–value
+database, cache and message broker, with optional durability. In their talk,
+Allen details the unification of Redis and Trino and demonstrate Redis' new
+connector for Trino can push down advanced queries and aggregations while
+leveraging Redis' unique in-memory secondary indexing.
+
+<!--more-->
+
+{% youtube JjBtZ26IHYk %}
+
+## Recap
+
+Redis is an open source, in-memory, noSQL database, which natively supports a
+variety of data structures and data models each natively designed for utmost
+performance and high throughput use cases across different types of workloads.
+Redis is widely known for being the fastest data store in the market with sub
+millisecond performance, its ease of use, and being a multi-model database.
+Redis is able to map relational tables to a key value database by adding a key
+value pair as a hash attribute for each column. However, how can you search for
+a certain key in a way that scales well in high throughput databases? Redis has
+a unique way to deal with this problem: secondary indexing and Redis Search.
+
+Redis Search enables acquiring secondary indexing and full text search allowing
+Redis to support multi-field queries, aggregations, and much other capabilities.
+As Allen says, "Redis Search will be at the heart of our new integration with
+Trino and game-changing better performance at scale to the existing Redis Trino
+connector". In addition, Redis supports a native data model for JSON documents,
+allowing you to store, update, and retrieve JSON values in a Redis database like
+other Redis data types. It also works with Redis Search to let you index and
+query JSON documents.
+
+The syntax for Redis Search is a bit different from traditional SQL syntax, so
+they have a Redis Trino connector that lets you easily integrate with
```

Comment:
```suggestion
they have a Redis-Trino connector that lets you easily integrate with
```
We should expand on this. There is _already_ a Redis connector in Trino. They're contributing an improved, much faster, more reliable rewrite of the connector to replace it.